### PR TITLE
Move AddProjectsDialog to projects module

### PR DESCRIPTION
**Pull request by seadowg**
This was something we discussed as a follow-up to #4519. I ended up moving a few helpers and shared resources android (mostly to `androidshared` and `testshared`) so it's probably worth going commit by commit on this one.

#### What has been done to verify that this works as intended?

Ran tests and verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

Only potentially interesting thing here is the Dagger setup I've added for `projects` but this just follows the pattern we introduced for `audiorecorder`.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

This can skip QA as it just moves some code around.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/build.gradle [hunk 1]
```diff
@@ -361,8 +361,12 @@ dependencies {
     testImplementation "org.mockito:mockito-core:3.7.7"
     testImplementation 'org.mockito:mockito-inline:3.7.7'
 
-    testImplementation project(path: ':testshared')
-    testImplementation "org.robolectric:robolectric:4.3.1"
+    testImplementation(project(path: ':testshared')) {
+        exclude group: 'org.robolectric' // Some tests in `collect_app` don't work with newer Robolectric
```
Comment (seadowg) [inline collect_app/build.gradle:365]: This problem is best left to our next "update all the libraries" cycle as it will involve reworking a few unrelated tests.

File: collect_app/src/main/res/layout/material_full_screen_dialog_toolbar.xml [hunk 2]
```diff
@@ -1,17 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<com.google.android.material.appbar.AppBarLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (grzesiek2010) [inline collect_app/src/main/res/layout/material_full_screen_dialog_toolbar.xml:2]: Why did you remove this layout? Couldn't you just change that style here?

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

**Comment by grzesiek2010** [comment]
Please fix conflicts.

File: collect_app/src/main/res/layout/material_full_screen_dialog_toolbar.xml (hunk 2 above)
Comment (seadowg) [reply collect_app/src/main/res/layout/material_full_screen_dialog_toolbar.xml:2]: I played around with moving it around to a shared module but in the end it seemed simpler to just have a shared style (`Widget.AndroidShared.Toolbar.Primary.CloseIcon`). I find `includes` can be a little awkward as you can't then customize/override attributes of the shared layout from where they are being used.

**Review by grzesiek2010 — APPROVED** [review]
